nft-example/src/lib.rs

```diff
@@ -0,0 +1,232 @@
+#![no_std]
+#![feature(const_btree_new)]
+
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::{H256, U256};
+use scale_info::TypeInfo;
+
+pub mod payloads;
+pub use payloads::{ApproveForAllInput, ApproveInput, InitConfig, TransferInput};
+
+pub mod state;
+pub use state::{State, StateReply};
+
+use non_fungible_token::base::NonFungibleTokenBase;
+use non_fungible_token::{Approve, ApproveForAll, NonFungibleToken, Transfer};
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new(H256::zero().to_fixed_bytes());
+
+#[derive(Debug, Decode, TypeInfo)]
+pub enum Action {
+    Mint,
+    Burn(U256),
+    Transfer(TransferInput),
+    Approve(ApproveInput),
+    ApproveForAll(ApproveForAllInput),
+    OwnerOf(U256),
+    BalanceOf(H256),
+}
+
+#[derive(Debug, Encode, Decode, TypeInfo)]
+pub enum Event {
+    Transfer(Transfer),
+    Approval(Approve),
+    ApprovalForAll(ApproveForAll),
+    OwnerOf(H256),
+    BalanceOf(U256),
+}
+
+#[derive(Debug)]
+pub struct NFT {
+    pub tokens: NonFungibleToken,
+    pub token_id: U256,
+    pub owner: ActorId,
+}
+
+static mut CONTRACT: NFT = NFT {
+    tokens: NonFungibleToken {
```

Comment:
Why is this named `tokens`? I would say `token`, or `token_interface` would make more sense.

Reply:
I thought about it and decided to name this variable in the plural, because it includes all minted tokens and information about them. But  probably it would be betted in single form